examples/nucleo_f401re/adc_dma/TimerHandler.hpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2023, Zühlke Engineering (Austria) GmbH
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef EXAMPLE_TimerHANDLER_HPP
+#define EXAMPLE_TimerHANDLER_HPP
+
+#include <modm/architecture/interface/interrupt.hpp>
+#include <modm/board.hpp>
+#include <modm/platform.hpp>
+
+#include "AdcDma.hpp"
+
+using namespace Board;
+using namespace modm::platform;
+using namespace std::chrono_literals;
+
+template<class Timer>
+void
+advancedTimerConfig(uint8_t repetitionCount)
+{
+	Timer::enable();
+	Timer::setMode(Timer::Mode::UpCounter, Timer::SlaveMode::Disabled,
+				   Timer::SlaveModeTrigger::Internal0, Timer::MasterMode::Update, true);
+	Timer::setPrescaler(84);
+	Timer::setOverflow(9999);
+	Timer::enableInterruptVector(Timer::Interrupt::CaptureCompare1, true, 0);
+	Timer::enableInterrupt(Timer::Interrupt::CaptureCompare1);
+	Timer::setRepetitionCount(repetitionCount);
+
+	Timer::enableOutput();
+	Timer::configureOutputChannel(1, Timer::OutputCompareMode::Pwm, 999, Timer::PinState::Enable);
```

Comment:
The interrupt not much, I was using it just to show the user that the timer is working. The output channel is important because I am using its event (when TIM1->CNT reaches 999) to trigger the adc.  This timer is set ti produce 50 pulses, just enough to fill the dma buffer.

Reply:
Sorry, I see it now. Before I've looked it up in the reference manual I assumed `Adc1::RegularConversionExternalTrigger::Event0` was the timer trigger output, but it is actually the channel 1 capture compare event. Could you add a comment when configuring `Adc1::enableRegularConversionExternalTrigger()` what `Event0` means? That would make it easier to understand.